### PR TITLE
Idaho capital gains deduction

### LINKED ISSUE #2769: Idaho Capital Gain Deduction
reference: https://casetext.com/regulation/idaho-administrative-code/title-idapa-35-tax-commission-state/rule-350101-income-tax-administrative-rules/section-350101170-idaho-capital-gains-deduction-in-general

Tax form: https://tax.idaho.gov/wp-content/uploads/forms/EFO00093/EFO00093_05-19-2022.pdf

### PR DESCRIPTION
**Pull request by jiajiagong0123**
### Summary
📄🛠️🔌

This pull request adds the `id_capital_gains_deduction` variable to the policy engine for the state of Idaho, which allows filers to deduct a percentage of their net capital gains from their taxable income. It also adds the corresponding parameter, test, and changelog files.

> _`extensions.json`_
> _A placeholder for now_
> _Winter of coding_

### Walkthrough
*  Introduce a new variable for the Idaho Capital Gain Deduction ([link](https://github.com/PolicyEngine/policyengine-us/pull/2963/files?diff=unified&w=0#diff-dee30235a52cb449714367a5e403a00121a4c8f1af38f144bb707b591a216a80R1-R26))
*  Define the policy parameter for the percentage of net capital gain that can be deducted ([link](https://github.com/PolicyEngine/policyengine-us/pull/2963/files?diff=unified&w=0#diff-768ddbc89d75b4c6f979c7cd3f8e75e07ae48e5d37f124ae57a8f94f499171b3R1-R13))
*  Add two test cases to verify the deduction amount for different capital gain scenarios ([link](https://github.com/PolicyEngine/policyengine-us/pull/2963/files?diff=unified&w=0#diff-a34d8041b3afd588c1679f441cdbaf2e17b1b2eb3a4f48c5a21ec81d1871723bR1-R17))
*  Document the minor feature in the changelog entry ([link](https://github.com/PolicyEngine/policyengine-us/pull/2963/files?diff=unified&w=0#diff-5885042762833a4aef51405a4036067e4e6aefa7a5091a2987d1d9d1fc2647e8R1-R4))
*  Create an empty recommendations array for the VS Code editor ([link](https://github.com/PolicyEngine/policyengine-us/pull/2963/files?diff=unified&w=0#diff-c16655a98a3ee89a7636a59c59a72b0e93649e3a1e947327cfc43a1336b4e912L1-R2))

Fixes #2769

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Idaho Capital Gain Deduction.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Idaho capital gain deduction.
```

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+description: Idaho allows filers to deduct this percentage of their capital gains.
+values:
+  2021-01-01: 0.6
+metadata:
+  unit: /1
+  label: Idaho capital gains deduction percentage
+  reference:
+  - title: 2022 Idaho Tax form - Capital Gains Deduction
+    href: https://tax.idaho.gov/wp-content/uploads/forms/EFO00093/EFO00093_05-19-2022.pdf#page=1
+  - title: IDAHO CAPITAL GAINS DEDUCTION-Section 35.01.01.170- pg.1- 01.02
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml:10]: need to find a law reference that mentions the 60% - need a law

File: policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    definition_period = YEAR
+    reference = "https://casetext.com/regulation/idaho-administrative-code/title-idapa-35-tax-commission-state/rule-350101-income-tax-administrative-rules/section-350101170-idaho-capital-gains-deduction-in-general"
+    defined_for = StateCode.ID
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.id.tax.income.deductions.capital_gains
+        # taxpayer must report capital gain net income
+        net_capital_gain = tax_unit("net_capital_gain", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py:18]: actually, the deduction only applies to capital gains from all property tax, we need to make a new variable called property_tax_net_capital_gain (just a class an no formula) and use it here

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+description: Idaho allows filers to deduct this percentage of their capital gains.
+values:
+  2021-01-01: 0.6
+metadata:
+  unit: /1
+  label: Idaho capital gains deduction percentage
+  reference:
+  - title: 2022 Idaho Tax form - Capital Gains Deduction
+    href: https://tax.idaho.gov/wp-content/uploads/forms/EFO00093/EFO00093_05-19-2022.pdf#page=1
+  - title: Idaho Legislature 63-3022H.  DEDUCTION OF CAPITAL GAINS.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml:10]: add section

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+description: Idaho allows filers to deduct this percentage of their capital gains.
+values:
+  2021-01-01: 0.6
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml:3]: in 2000: 0.8
in 2001 and on: 0.6

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+description: Idaho allows filers to deduct this percentage of their capital gains.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml:1]:
```suggestion
description: Idaho allows filers to deduct this percentage of their capital gains on property sales.
```

File: policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py (hunk 3 above)
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py:18]:
```suggestion
        net_capital_gain = tax_unit("property_tax_net_capital_gain", period)
```

File: policyengine_us/tests/policy/baseline/gov/states/id/tax/income/deductions/id_capital_gains_deduction.yaml [hunk 7]
```diff
@@ -0,0 +1,17 @@
+- name: No capital gains
+  period: 2022
+  absolute_error_margin: 0.001
+  input:
+    net_capital_gain: 0
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/id/tax/income/deductions/id_capital_gains_deduction.yaml:5]:
```suggestion
    property_tax_net_capital_gain: 0
```

File: policyengine_us/tests/policy/baseline/gov/states/id/tax/income/deductions/id_capital_gains_deduction.yaml [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    state_code: ID
+  output:
+    id_capital_gains_deduction: 0
+
+- name: 60% net capital gains 
+  period: 2022 
+  absolute_error_margin: 0.001
+  input:
+    net_capital_gain: 6_000
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/id/tax/income/deductions/id_capital_gains_deduction.yaml:14]:
```suggestion
    property_tax_net_capital_gain: 6_000
```

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+values:
+  2000-01-01: 0.8
+  2001-01-01: 0.6
+metadata:
+  unit: /1
+  label: Idaho capital gains deduction percentage
+  reference:
+  - title: 2022 Idaho Tax form - Capital Gains Deduction
+    href: https://tax.idaho.gov/wp-content/uploads/forms/EFO00093/EFO00093_05-19-2022.pdf#page=1
+  - title: Idaho Legislature Section 63-3022H.  DEDUCTION OF CAPITAL GAINS.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml:11]: add specific part of legislation

File: policyengine_us/variables/gov/irs/tax/federal_income/capital_gains/property_tax_net_capital_gain.py [hunk 10]
```diff
@@ -0,0 +1,11 @@
+from policyengine_us.model_api import *
+
+
+class property_tax_net_capital_gain(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Idaho property tax net capital gain"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://casetext.com/regulation/idaho-administrative-code/title-idapa-35-tax-commission-state/rule-350101-income-tax-administrative-rules/section-350101170-idaho-capital-gains-deduction-in-general"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/irs/tax/federal_income/capital_gains/property_tax_net_capital_gain.py:10]:
```suggestion
```

File: policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py [hunk 11]
```diff
@@ -0,0 +1,26 @@
+from policyengine_us.model_api import *
+
+
+class id_capital_gains_deduction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Idaho capital gain deduction"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://casetext.com/regulation/idaho-administrative-code/title-idapa-35-tax-commission-state/rule-350101-income-tax-administrative-rules/section-350101170-idaho-capital-gains-deduction-in-general"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py:10]:
```suggestion
    reference = "https://legislature.idaho.gov/statutesrules/idstat/title63/t63ch30/sect63-3022h/"
```

File: policyengine_us/variables/gov/irs/tax/federal_income/capital_gains/property_tax_net_capital_gain.py [hunk 12]
```diff
@@ -0,0 +1,11 @@
+from policyengine_us.model_api import *
+
+
+class property_tax_net_capital_gain(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Idaho property tax net capital gain"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/irs/tax/federal_income/capital_gains/property_tax_net_capital_gain.py:7]:
```suggestion
    label = "Net capital gain from sale or exchange of property"
```

File: policyengine_us/variables/gov/irs/tax/federal_income/capital_gains/property_tax_net_capital_gain.py [hunk 13]
```diff
@@ -0,0 +1,11 @@
+from policyengine_us.model_api import *
+
+
+class property_tax_net_capital_gain(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/irs/tax/federal_income/capital_gains/property_tax_net_capital_gain.py:4]:
```suggestion
class property_sales_net_capital_gain(Variable):
```

File: policyengine_us/variables/gov/irs/tax/federal_income/capital_gains/property_tax_net_capital_gain.py [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+    value_type = float
+    entity = TaxUnit
+    label = "Idaho property tax net capital gain"
+    unit = USD
+    definition_period = YEAR
+    reference = "https://casetext.com/regulation/idaho-administrative-code/title-idapa-35-tax-commission-state/rule-350101-income-tax-administrative-rules/section-350101170-idaho-capital-gains-deduction-in-general"
+    defined_for = StateCode.ID
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/irs/tax/federal_income/capital_gains/property_tax_net_capital_gain.py:11]:
```suggestion
```

File: policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+    defined_for = StateCode.ID
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.id.tax.income.deductions.capital_gains
+        # taxpayer must report capital gain net income
+        net_capital_gain = tax_unit("property_tax_net_capital_gain", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py:18]:
```suggestion
        net_capital_gain = tax_unit("property_sales_net_capital_gain", period)
```

File: policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py [hunk 16]
```diff
@@ -0,0 +1,26 @@
+from policyengine_us.model_api import *
+
+
+class id_capital_gains_deduction(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "Idaho capital gain deduction"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py:7]:
```suggestion
    label = "Idaho capital gains deduction"
```

File: changelog_entry.yaml [hunk 17]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Idaho capital gain deduction.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Idaho capital gains deduction.
```

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/variables/gov/irs/tax/federal_income/capital_gains/property_tax_net_capital_gain.py
Comment (MaxGhenis) [inline policyengine_us/variables/gov/irs/tax/federal_income/capital_gains/property_tax_net_capital_gain.py:1]: move to household income folder (variables in the IRS folder are those computed according to IRS rules)

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml [hunk 18]
```diff
@@ -0,0 +1,14 @@
+description: Idaho allows filers to deduct this percentage of their capital gains on property sales.
+values:
+  2000-01-01: 0.8
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml:3]: Isn't it 2001, then 2002?

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml [hunk 19]
```diff
@@ -0,0 +1,14 @@
+description: Idaho allows filers to deduct this percentage of their capital gains on property sales.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/capital_gains/percentage.yaml:1]:
```suggestion
description: Idaho allows filers to deduct this fraction of capital gains on qualified property sales.
```

File: policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py [hunk 20]
```diff
@@ -0,0 +1,26 @@
...
+    defined_for = StateCode.ID
+
+    def formula(tax_unit, period, parameters):
+        p = parameters(
+            period
+        ).gov.states.id.tax.income.deductions.capital_gains
+        # taxpayer must report capital gain net income
+        net_capital_gain = tax_unit("property_sales_net_capital_gain", period)
+
+        # Filers can deduct 60% of captial gains
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/id/tax/income/deductions/id_capital_gains_deduction.py:20]: Delete - avoid including parameter values here

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/household/income/tax_unit/property_tax_net_capital_gain.py [hunk 21]
```diff
@@ -0,0 +1,9 @@
+from policyengine_us.model_api import *
```
Comment (MaxGhenis) [inline policyengine_us/variables/household/income/tax_unit/property_tax_net_capital_gain.py:1]: Rename file to align with variable

**Review by MaxGhenis — APPROVED** [review]
Good to merge after this

**Pull request merged**